app/components/validation/card.js

```diff
@@ -0,0 +1,229 @@
+import Component from '@glimmer/component';
+
+import { guidFor } from '@ember/object/internals';
+import { tracked } from '@glimmer/tracking';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import { SHACL, SKOS, RDF, FORM } from '@lblod/submission-form-helpers';
+import { ForkingStore } from '@lblod/ember-submission-form-fields';
+import { restartableTask, timeout } from 'ember-concurrency';
+import { Literal, Namespace } from 'rdflib';
+import { getPrefLabelOfNode } from '../../utils/forking-store-helpers';
+import { sortObjectsOnProperty } from '../../utils/sort-object-on-property';
+import { isValidationConfigValidForType } from '../../utils/validation/is-validation-config-valid-for-type.js';
+
+const EXT = new Namespace('http://mu.semte.ch/vocabularies/ext/');
+
+export default class ValidationCardComponent extends Component {
+  inputId = 'validation-card' + guidFor(this);
+
+  @service intl;
+
+  @tracked validation;
+  @tracked validationType;
+  @tracked validationSubject;
+
+  @tracked validationTypes;
+
+  @tracked defaultErrorMessage;
+  @tracked isCardInError;
+
+  metaStore;
+
+  constructor() {
+    super(...arguments);
+
+    this.validation = this.args.validation;
+    this.validationSubject = this.validation.subject;
+    this.setup.perform();
+  }
+
+  getGroupingForValidation(typeNode) {
+    let grouping = FORM('grouping');
+    const groupingFromMeta = this.metaStore.any(
+      typeNode,
+      FORM('grouping'),
+      undefined,
+      this.metaGraph
+    );
+
+    if (groupingFromMeta) {
+      grouping = {
+        object: groupingFromMeta,
+        predicate: FORM('grouping'),
+      };
+    }
+
+    return grouping;
+  }
+
+  getDefaultErrorMessage() {
+    let message = '';
+    if (this.validationType) {
+      const messageLiteral = this.metaStore.any(
+        this.validationType.object,
+        SHACL('resultMessage'),
+        undefined,
+        this.metaGraph
+      );
+
+      if (messageLiteral) {
+        message = messageLiteral.value;
```

Comment:
Not that important I guess, but the language tag will not be used

Reply:
yeah but does it hurt to add it. I mean it is given in the meta.ttl

Reply:
What would you be adding? I can think of a selector component that will show the options `@nl` and `@en` but would you use it and is it really an addition?

Reply:
Ah I see sorry, the default message that is being added has no language tag. They are all @nl I would leave it. Like I said above if users change the result message it is not added and when they don't there is only one message that can be used

Reply:
I was thinking only in case of default message, as it is given this way. Giving an extra selector for the user is maybe overkill.

Reply:
If you like you can create an analyse ticket for this and link it to SFB-229 🙂